### DISCUSSION TITLE
About splitting Mobile / Desktop UI

### DISCUSSION THREAD

**Discussion by sfiannaca** [Mobile-Artificial-Intelligence/maid#71] [Ideas]
I like the constraints approach rather than checking the platform because the user can use it on desktop as mobile if needed.
I can try to help you with that but i dont want to mess up your folders projects.
Do you have any ideas?

I like something like that: 
- ui
  - mobile
  - desktop
  - shared

It's almost twice the amount of code, but I believe it's worth it.
For example Drawer can be moved on shared if used on both layout

```
return Scaffold(
      appBar: AppBar(title: const Text('LayoutBuilder Example')),
      body: LayoutBuilder(
        builder: (BuildContext context, BoxConstraints constraints) {
          if (constraints.maxWidth > 600) {
            return _buildDesktopWidget();
          } else {
            return _buildMobileWidget();
          }
        },
      ),
    );
```

**Comment by danemadsen** [comment]
Im going to expose the ui switcher as a setting when i get around to it
